In [1]:
#DEPENDENCIES & INITIALISATIONS
from IPython.display import HTML #For embedding videos
import math #Basic math functionality
import numpy as np #Numpy for working with arrays
import matplotlib.pyplot as plt #Plotting functionality 
#HTML('<center><iframe width="560" height="315" src="#" frameborder="0" allowfullscreen></iframe></center>')
print("- Dependencies imported 👍")
print("- Remember to select 'Kernel-> Restart & Run All' on first load to show embedded Youtube videos 🎬")
print("- Author: Dr H Alwi & Dr SPCarroll, Sept 2020, Rev 0")

- Dependencies imported 👍
- Remember to select 'Kernel-> Restart & Run All' on first load to show embedded Youtube videos 🎬
- Author: Dr H Alwi & Dr SPCarroll, Sept 2020, Rev 0


# Week 7: Polar Coordinates & Force, Mass and Acceleration
Reference: Chapter 13 of Engineering Mechanics - Statics and Dynamics Principles, Bedford A. & Fowler, W., 2003

## Question 1

A robotic arm is programmed so that the point $P$ (manipulator) traverses a path described by the polar coordinates,

\begin{equation*}
r=1-0.5\cos(2\pi t) \:m
\end{equation*}

\begin{equation*}
\theta = 0.5-0.2\sin(2\pi t) \:rad
\end{equation*}

At $t=0.8\:s$ determine:
- (a) the velocity of $P$ in terms of polar coordinates.
- (b) the cartesian components of the velocity of $P$.

![title](img/T2W7/img1.jpg)

**Solution 1**

(a) We've seen that the velocity in terms of polar coordinates is given by,

\begin{equation*}
\mathbf{v} = \frac{\mathrm{d}r}{\mathrm{d}t}\mathbf{e}_r + r \frac{\mathrm{d}\theta}{\mathrm{d}t}\mathbf{e}_{\theta}
\end{equation*}

So we can simply differentiate the expressions for $r$ and $\theta$ provided and substitute them into the equation above to obtain the velocity in polar coordinates. 

\begin{equation*}
\frac{\mathrm{d}r}{\mathrm{d}t} = \pi\sin(2\pi t)
\end{equation*}

\begin{equation*}
\frac{\mathrm{d}\theta}{\mathrm{d}t} = -0.4\pi\cos(2\pi t)
\end{equation*}

Therefore the velocity as a function of time in polar coordinates is given by,

\begin{equation*}
\mathbf{v} = \pi\sin(2\pi t)\:\mathbf{e}_r + (1-0.5\cos(2\pi t))(-0.4\pi\cos(2\pi t))\:\mathbf{e}_\theta
\end{equation*}

Evaluating at $t=0.8$ seconds,

In [2]:
pi=math.pi #Redine pi for convenience
t=0.8 #(s) Time
v_radial = pi*math.sin(2*pi*t) #(m/s) Magnitude of radial component of velocity
v_trans = (1-0.5*math.cos(2*pi*t))*(-0.4*pi*math.cos(2*pi*t))
print('The magnitude of the radial component of velocity is {one} m/s'.format(one=round(v_radial,2)))
print('The magnitude of the transverse component of velocity is {one} m/s'.format(one=round(v_trans,2)))

The magnitude of the radial component of velocity is -2.99 m/s
The magnitude of the transverse component of velocity is -0.33 m/s


Therefore the complete velocity vector in polar coordinates at $t=0.8$ seconds is,

\begin{equation*}
\mathbf{v} = -2.99\mathbf{e}_r -0.33\mathbf{e}_\theta\:m/s
\end{equation*}

(b) We can obtain the cartesian components using simple geometry. First we obtain the angle $\theta$ at $t=0.8$ seconds by directly substituting into the equation provided for $\theta$,

In [3]:
theta = 0.5-0.2*math.sin(2*pi*t) #(rads) Angular position of the robot arm
print('The angular position of the robot arm is {one} radians.'.format(one=round(theta,3)))

The angular position of the robot arm is 0.69 radians.


Now we simply resolve the radial and transverse velocity vectors into their $x$ and $y$ components.

![title](img/T2W7/img4.jpg)

\begin{equation*}
v_x = v_{x,r} + v_{x,\theta}
\end{equation*}

\begin{equation*}
v_y = v_{y,r} + v_{y,\theta}
\end{equation*}

Evaluating at $t=0.8$ seconds,

In [4]:
vx = v_radial*math.cos(theta)-v_trans*math.sin(theta)
vy = v_radial*math.sin(theta) + v_trans*math.cos(theta)
print('The x and y components of the velocity are {one} and {two} m/s respectively'.format(one=round(vx,2),two=round(vy,2)))

The x and y components of the velocity are -2.09 and -2.16 m/s respectively


We can confirm our solution by making sure that the magnitude of the velocity in cartesian coordinates is the same as the magnitude obtained for polar coordinates. Remember, although the coordinate reference frame might change we are still describing the velocity of the same point so its velocity should be the same regardless of the reference frame used.

In [5]:
V_polar = ((v_radial**2)+(v_trans**2))**0.5
V_cart = ((vx**2)+(vy**2))**0.5
print('The magnitude of the velocity in polar coordinates is {one} m/s'.format(one=round(V_polar,2)))
print('The magnitude of the velocity in cartesian coordinates is {one} m/s'.format(one=round(V_cart,2)))

The magnitude of the velocity in polar coordinates is 3.01 m/s
The magnitude of the velocity in cartesian coordinates is 3.01 m/s


## Question 2


The polar coordinates of the collar $A$ are given as functions of time,

\begin{equation*}
r=1+0.2t^2\:m
\end{equation*}

\begin{equation*}
\theta=2t \:rad
\end{equation*}

What are the magnitudes of the velocity and acceleration of the collar at $t=2$ seconds?

![title](img/T2W7/img2.jpg)

**Solution 2**

We can start by following the same procedure as question 1 above to determine the velocity magnitude at $t=2$ seconds. Evaluating the differential of $r$ and $\theta$ yields,

\begin{equation*}
\frac{\mathrm{d}r}{\mathrm{d}t} = 0.4t
\end{equation*}

\begin{equation*}
\frac{\mathrm{d}\theta}{\mathrm{d}t} = 2
\end{equation*}

Noting that the polar velocity vector is,

\begin{equation*}
\mathbf{v} = \frac{\mathrm{d}r}{\mathrm{d}t}\mathbf{e}_r + r \frac{\mathrm{d}\theta}{\mathrm{d}t}\mathbf{e}_{\theta}
\end{equation*}

We can substitute into this equation to obtain, 

\begin{equation*}
\mathbf{v} = 0.4t\:\mathbf{e}_r + (2+0.4t^2)\mathbf{e}_\theta
\end{equation*}

At $t=2$ seconds we have,

\begin{equation*}
\mathbf{v} = 0.8\:\mathbf{e}_r + 3.6\:\mathbf{e}_\theta\:m/s
\end{equation*}

Therefore we can evaluate the velocity magnitude as,

\begin{equation*}
v=\sqrt{(0.8^2 + 3.6^2)}
\end{equation*}

In [6]:
v = math.sqrt(0.8**2+3.6**2)
print('(a) The velocity magnitude is {v} m/s'.format(v=round(v,2)))

(a) The velocity magnitude is 3.69 m/s


To obtain the acceleration we first note the expression,

\begin{equation*}
\mathbf{a} = \underbrace{\left[\frac{\mathrm{d}^2r}{\mathrm{d}t^2} - r\left( \frac{\mathrm{d}\theta}{\mathrm{d}t}\right)^2\right]}_{a_r}\mathbf{e}_r + \underbrace{\left[r\frac{\mathrm{d}^2\theta}{\mathrm{d}t^2} +2 \frac{\mathrm{d}r}{\mathrm{d}t}\frac{\mathrm{d}\theta}{\mathrm{d}t} \right]}_{a_\theta}\mathbf{e}_\theta
\end{equation*}

Evaluating the second differential of $r$ and $\theta$,

\begin{equation*}
\frac{\mathrm{d}^2r}{\mathrm{d}t^2} = 0.4
\end{equation*}

\begin{equation*}
\frac{\mathrm{d}^2\theta}{\mathrm{d}t^2} = 0
\end{equation*}

Now evaluating $a_r$ and $a_\theta$,

\begin{equation*}
a_r = 0.4-(1+0.2t^2)(2)^2
\end{equation*}

\begin{equation*}
a_\theta = (1+0.2t^2)(0) + 2(0.4t)(2)
\end{equation*}

Evaluating at $t=2$ seconds,

In [7]:
t=2 #(s) Time
ar = 0.4-(1+0.2*t**2)*(2)**2 #(m/s^2) Magnitude of radial component of acceleration
at = 2*0.4*t*2 #(m/s^2) Magnitude of transvers component of acceleration
a = math.sqrt(ar**2+at**2)  #(m/s^2) Magnitude of resultant acceleration vector

print('(b) The acceleration magnitude is {one} m/s^2'.format(one=round(a,2)))

(b) The acceleration magnitude is 7.52 m/s^2


## Question 3

The radial coordinate of a satellite is related to its angular position $\theta$ by the following equation,

\begin{equation*}
r=\frac{1.91\times 10^7}{1+0.5\cos\theta}\:m
\end{equation*}

The product of the radial position and the transverse component of the velocity is,

\begin{equation*}
rv_\theta = 8.72\times 10^{10}\:m^2/s
\end{equation*}

What is the satellite's velocity in terms of polar coordinates when $\theta=90^\circ$?

![title](img/T2W7/img3.jpg)

**Solution**

In this example we are given the radial position $r$ as a function of $\theta$ but in order to evaluate the polar velocity vector we need the differential of $r$ with respect to time, therefore,

\begin{equation*}
\frac{\mathrm{d}r}{\mathrm{d}t} = \frac{\mathrm{d}r}{\mathrm{d}\theta}\frac{\mathrm{d}\theta}{\mathrm{d}t}
\end{equation*}

We start by evaluating $\frac{\mathrm{d}r}{\mathrm{d}\theta}$ by differentiating the expression provided in the question,

In [8]:
import sympy as sym #Import SymPy

#Define some symbols 
A = sym.Symbol('A') #A constant to temporarily replace 1.91*10^7 (make things easier to read)
T = sym.Symbol('T') #Symbol for theta
r = sym.Symbol('r') #Symbol for r

#Construct the function to differentiate
r = A/(1+0.5*sym.cos(T))

drdT = sym.diff(r,T) #Differentiate r with respect to theta
drdT = sym.simplify(drdT) #Try to simplify the resulting expression
sym.pprint(drdT) #Print the result in an easy to read format

   0.5⋅A⋅sin(T)  
─────────────────
                2
(0.5⋅cos(T) + 1) 


Therefore we have,

\begin{equation*}
\frac{\mathrm{d}r}{\mathrm{d}\theta} = \frac{0.5A\sin\theta}{(0.5\cos\theta+1)^2}
\end{equation*}

where $A=1.91\times 10^7$. Evaluating $\frac{\mathrm{d}r}{\mathrm{d}\theta}$,

In [9]:
theta=pi/2 #(rad)
A = 1.91*10**7 # constant
drdT = (0.5*A*math.sin(theta))/(0.5*math.cos(theta)+1)**2
print('The differential of r with respect to theta is {one}'.format(one=round(drdT)))

The differential of r with respect to theta is 9550000


Next we evaluate $\frac{\mathrm{d}\theta}{\mathrm{d}t}$ by noting from the question,

\begin{equation*}
rv_{\theta} = r^2\frac{\mathrm{d}\theta}{\mathrm{d}t} = 8.72\times10^{10}
\end{equation*}

When $\theta=90^\circ$ we evaluate $r$ and then solve for $\frac{\mathrm{d}\theta}{\mathrm{d}t}$,

In [10]:
r = A/(1+0.5*math.cos(theta)) #(m) radius when theta = 90 degrees
dTdt = (8.72*10**10)/r**2
print('When theta is 90 degrees, r is {one} m'.format(one=r))
print('The differential of theta with respect to time is {one} rad/s'.format(one=round(dTdt,6)))

When theta is 90 degrees, r is 19100000.0 m
The differential of theta with respect to time is 0.000239 rad/s


Finally we can evaluate the polar velocity vector,

\begin{equation*}
\mathbf{v} = \frac{\mathrm{d}r}{\mathrm{d}t}\mathbf{e}_r + r \frac{\mathrm{d}\theta}{\mathrm{d}t}\mathbf{e}_{\theta}
\end{equation*}

\begin{equation*}
\mathbf{v} = (9550000\times 0.000239)\:\mathbf{e}_r + (19100000\times 0.000239)\:\mathbf{e}_\theta
\end{equation*}

\begin{equation*}
\mathbf{v} = 2282\:\mathbf{e}_r + 4565\:\mathbf{e}_\theta \:m/s
\end{equation*}

## Question 4

The total force exerted on an $80000\:N$ rocket during launch, including engine thrust, aerodynamic drag and self-weight between $2\leq t \leq4$ is given by the function,

\begin{equation*}
\sum\mathbf{F} = (2000-400t^2)\mathbf{i} +(5200+440t)\mathbf{j} + (800+60t^2)\mathbf{k}\:N
\end{equation*}

At $t=2$ seconds, its velocity is,

\begin{equation*}
\mathbf{v} = 12\mathbf{i} + 220\mathbf{j}-30\mathbf{k}\:m/s
\end{equation*}

What is the velocity at $t=4$ seconds?

**Solution**

The equation of motion is $\mathbf{F} = m\:\mathbf{a}$. So we can solve for acceleration and them integrate the acceleration between $2\leq t\leq4$ to obtain the velocity. To process through the solution, we consider motion in each direction independently, i.e. break the motion into its components.

The mass of the rocket is,

\begin{equation*}
m = \frac{80000}{9.81} = 8154.7\:kg
\end{equation*}

**x-direction**

\begin{equation*}
a_x =\frac{\mathrm{d}v_x}{\mathrm{d}t} = \frac{F_x}{m} = \frac{2000-400t^2}{8154.7} = 0.245-0.05t^2 \:m/s^2
\end{equation*}

Noting the initial velocity in the $x$-direction and integrating between the limits,

\begin{equation*}
\int_{12}^v\mathrm{d}v_x = \int_2^4 (0.245-0.05t^2) \mathrm{d}t = 11.56\:m/s
\end{equation*}

**y-direction**

\begin{equation*}
a_y =\frac{\mathrm{d}v_y}{\mathrm{d}t} = \frac{F_y}{m} = \frac{5200+440t}{8154.7} = 0.638+0.054t \:m/s^2
\end{equation*}

Integrating between the limits,

\begin{equation*}
\int_{220}^v\mathrm{d}v_y = \int_2^4 (0.638+0.054t) \mathrm{d}t = 222.8\:m/s
\end{equation*}

**z-direction**

\begin{equation*}
a_z =\frac{\mathrm{d}v_z}{\mathrm{d}t} = \frac{F_z}{m} = \frac{800+60t^2}{8154.7} = 0.098+0.0074t^2 \:m/s^2
\end{equation*}

Integrating between the limits,

\begin{equation*}
\int_{-30}^v\mathrm{d}v_z = \int_2^4 (0.098+0.0074t^2) \mathrm{d}t = -29.7\:m/s
\end{equation*}

Therefore the velocity vector is,

\begin{equation*}
\mathbf{v} = (11.56\:\mathbf{i} + 222.8\:\mathbf{j}-29.7\:\mathbf{k})\:m/s
\end{equation*}

## Answers

1. $\mathbf{v} = -2.99\mathbf{e}_r -0.33\mathbf{e}_\theta\:m/s$, $v_x = -2.09\:m/s$ $v_y=-2.16 \:m/s $
2. $3.69 \:m/s$, $7.52 \:m/s^2$
3. $\mathbf{v} = 2282\:\mathbf{e}_r + 4565\:\mathbf{e}_\theta \:m/s$
4. $\mathbf{v} = (11.56\:\mathbf{i} + 222.8\:\mathbf{j}-29.7\:\mathbf{k})\:m/s$